tests/console/thumbnail_test.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+require_once dirname(__FILE__) . '/../../phpBB/includes/functions_compatibility.php';
+require_once dirname(__FILE__) . '/../../phpBB/includes/functions.php';
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+use phpbb\console\command\thumbnail\generate;
+use phpbb\console\command\thumbnail\delete;
+use phpbb\console\command\thumbnail\recreate;
+
+class phpbb_console_command_thumbnail_test extends phpbb_database_test_case
+{
+	protected $db;
+	protected $config;
+	protected $cache;
+	protected $user;
+	protected $phpEx;
+	protected $phpbb_root_path;
+	protected $application;
+
+	public function getDataSet()
+	{
+		return $this->createXMLDataSet(dirname(__FILE__) . '/fixtures/thumbnail.xml');
+	}
+
+	public function setUp()
+	{
+		global $config, $phpbb_root_path, $phpEx, $phpbb_filesystem;
+
+		parent::setUp();
+
+		$config = $this->config = new \phpbb\config\config(array(
+			'img_min_thumb_filesize' => 2,
+			'img_max_thumb_width' => 2,
+			'img_imagick' => '',
+		));
+
+		$this->db = $this->db = $this->new_dbal();
+		$this->user = $this->getMock('\phpbb\user', array(), array(
+				new \phpbb\language\language(new \phpbb\language\language_file_loader($phpbb_root_path, $phpEx)),
+				'\phpbb\datetime')
+		);
+		$this->phpbb_root_path = $phpbb_root_path;
+		$this->phpEx = $phpEx;
+
+		$this->cache = $this->getMock('\phpbb\cache\service', array(), array(new phpbb_mock_cache(), $this->config, $this->db, $this->phpbb_root_path, $this->phpEx));
+		$this->cache->expects($this->any())->method('obtain_attach_extensions')->will($this->returnValue(array(
+			'png' => array('display_cat' => ATTACHMENT_CATEGORY_IMAGE),
+			'txt' => array('display_cat' => ATTACHMENT_CATEGORY_NONE),
+		)));
+
+		$this->application = new Application();
+		$this->application->add(new generate($this->user, $this->db, $this->cache, $this->phpbb_root_path, $this->phpEx));
+		$this->application->add(new delete($this->user, $this->db, $this->phpbb_root_path));
+		$this->application->add(new recreate($this->user));
+
+		$phpbb_filesystem = new \phpbb\filesystem\filesystem();
+	}
+
+	public function test_thumbnails()
+	{
+		copy(dirname(__FILE__) . '/fixtures/png.png', $this->phpbb_root_path . 'files/test_png_1');
```

Comment:
Should probably do this in the setUp method and remove those images in the tearDown method.
Also, you should be able to use dataproviders for this instead of repeating yourself.